proposals/p5270.md

```diff
@@ -0,0 +1,132 @@
+# Move explorer out of toolchain git repository
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/5270)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [Use a branch to refer to the explorer codebase](#use-a-branch-to-refer-to-the-explorer-codebase)
+    -   [Do nothing](#do-nothing)
+
+<!-- tocstop -->
+
+## Abstract
+
+The explorer codebase is no longer actively developed or maintained, but retains
+value as a place to look for implementations of various parts of the design.
+Having both the explorer and toolchain in the working git repository causes
+confusion for developers and users, and increases our incremental maintenance
+burden. We will move explorer out of the working git repository so that it
+remains accessible but is clearly deliniated from the toolchain and is more
+clearly frozen or archived.
+
+## Problem
+
+Users have expressed confusion when they come across examples or tests for the
+explorer that involve `Main()`, whereas the design and implementation of the
+toolchain have moved on to `Run()` as the entry point.
+
+Developers have attempted to update the explorer codebase to fix style guide
+issues[^1], as tools like grep or clang-tidy do not differentiate our active
+toolchain codebase from the frozen explorer one.
+
+As we update Bazel, Clang, Clang-tidy, etc, we also have to update the explorer
+codebase to keep it building cleanly.
+
+As we modify the `//testing` harness, we have to accommodate and work around the
+explorer tests[^2][^3][^4].
+
+[^1]: https://github.com/carbon-language/carbon-lang/pull/5224
+
+[^2]: https://github.com/carbon-language/carbon-lang/pull/4979
+
+[^3]: https://github.com/carbon-language/carbon-lang/pull/5025
+
+[^4]: https://github.com/carbon-language/carbon-lang/pull/5036
+
+## Background
+
+-   Previous proposal:
+    [p3532: Focus implementation effort on the toolchain](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p3532.md)
+-   Kick-off of this discussion
+    [#5224](https://github.com/carbon-language/carbon-lang/pull/5224#pullrequestreview-2730512195)
+-   Discord discussion:
+    [2025-04-02 #explorer](https://discord.com/channels/655572317891461132/763516049710120960/1356979197070803095)
+
+## Proposal
+
+1. Add a tag `explorer-archived` in the main `carbon-lang` git repository.
+2. Copy the `//explorer` directory and its dependencies to a new `explorer`
```

Comment:
Should `//installers` be included in this, since that's explorer-specific? It's not a dependency of `//explorer`, but it's only there for Compiler Explorer's build of `//explorer`.

Reply:
Yep, thanks, done. I will also include it in the sibling repo since it exists only for explorer. That way we just preserve whatever we have at head.